src/librustdoc/clean/utils.rs

```diff
@@ -171,11 +171,20 @@ crate fn get_real_types(
     cx: &DocContext<'_>,
     recurse: i32,
 ) -> FxHashSet<(Type, TypeKind)> {
+    fn insert(res: &mut FxHashSet<(Type, TypeKind)>, cx: &DocContext<'_>, ty: Type) {
+        if let Some(kind) = ty.def_id().map(|did| cx.tcx.def_kind(did).clean(cx)) {
+            res.insert((ty, kind));
+        } else {
+            // This is a primitive, let's store it as such.
```

Comment:
This isn't always a primitive. `ty.def_id()` will return `None` for generic parameters as well which shouldn't end up the in the search index.

Reply:
Great catch! I'll fix that and add a test to ensure they're not listed.

Reply:
Actually no because we check before if it's a full generic. So it should never be the case. I'll add a test to ensure it though.

Reply:
`is_full_generic` will only return true for generics passed by value. Generics passed by reference like `pub fn foo<T>(x: &T) {}` or `&self` show up in the search index with this PR.

Reply:
I see. I'll add a `is_primitive` method then to ensure that.